### ISSUE TITLE
[EKS]: Service Linked Role for Amazon EKS

### ISSUE THREAD

**Issue by tabern** [aws/containers-roadmap#243]
EKS will automatically provision a managed IAM role ([Service Linked Role](https://docs.aws.amazon.com/IAM/latest/UserGuide/using-service-linked-roles.html)) during cluster creation. This makes it simpler and faster to setup an EKS cluster. 

Helps solve https://github.com/aws/containers-roadmap/issues/44 

**Which service(s) is this request for?**
Amazon EKS

**Are you currently working around this issue?**
Manually create and manage a service IAM role per https://docs.aws.amazon.com/eks/latest/userguide/service_IAM_role.html

**Comment by mikestef9** [comment]
Wanted to provide an update on the status for the EKS service linked role.

We are planning to introduce a service-linked role that includes the permissions that EKS requires to create and manage clusters. Amazon EKS will not allow you to edit the capabilities of the service-linked role.

After we launch the feature, newly created EKS clusters will use this service-linked role. Under most circumstances, you don't need to manually create the service-linked role. When you create EKS clusters in the AWS Management Console, the AWS CLI, or the AWS API, Amazon EKS creates the service-linked role for you. If you delete this service-linked role, and then need to create it again, you can use the same process to recreate the role in your account. You can delete the Amazon EKS service-linked role only after first deleting your Amazon EKS clusters. 

*Note:* You must configure permissions to allow the IAM entity (such as a user, group, or role) that creates your clusters to create a service-linked role. For more information, see [Service-Linked Role Permissions](https://docs.aws.amazon.com/IAM/latest/UserGuide/using-service-linked-roles.html#service-linked-role-permissions) in the IAM User Guide. 

The Amazon EKS service-linked role protects you from inadvertently revoking Amazon EKS's required permissions to manage clusters on your behalf.

After launch, the [roleArn](https://docs.aws.amazon.com/eks/latest/APIReference/API_CreateCluster.html) required during EKS cluster creation will now only need to include the AmazonEKSClusterPolicy.

We are continuing to explore other ways to reduce friction during cluster creation that help solve [#44](https://github.com/aws/containers-roadmap/issues/44).

For more information on service linked roles, you can read the [IAM documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/using-service-linked-roles.html).

**Comment by mikestef9** [comment]
The EKS Service Linked Role is now live and will be used for all new Amazon EKS clusters. 

Note that for new clusters, only the AmazonEKSClusterPolicy needs to be attached to the role provided with cluster creation. The AmazonEKSServicePolicy is no longer needed, as those permissions are now included with the service linked role. 

Today, the service linked role will only be used with new clusters, and we are evaluating options for allowing existing clusters to be updated to use the SLR.

To learn more

- [What's New](https://aws.amazon.com/about-aws/whats-new/2020/04/amazon-eks-supports-service-linked-roles/)
- [Documentation](https://docs.aws.amazon.com/eks/latest/userguide/using-service-linked-roles-eks.html)

**Comment by SayakMukhopadhyay** [comment]
@mikestef9 apologies for necroing this issue but I thought I would rather ask here than create a new issue. I am getting onboarded to AWS and was wondering why the `AmazonEKSClusterPolicy` was not included in the Service-Linked roles getting added. I mean, before solving this issue, we had to create 2 roles with 2 policies. Why was `AmazonEKSClusterPolicy` still left to be manually assigned?

**Comment by mikestef9** [comment]
Before this feature you had to create one role with two policies attached. Now you only need to attach the `AmazonEKSClusterPolicy` to the role passed in during cluster creation. It's mainly some internal operational reasons that can't give full details here why we weren't able to move this policy to the SLR as well. It's something in the future we'd like to eventually do, but not on near term roadmap.
